client/lib/url/decode-utils.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ * Internal dependencies
+ */
+import { URL as TypedURL } from 'types';
+
+interface Stringable {
+	toString: () => string;
+}
+
+/**
+ * Wrap decodeURI in a try / catch block to prevent `URIError` on invalid input
+ * Passing a non-string value will return an empty string.
+ * @param  encodedURI URI to attempt to decode
+ * @return            Decoded URI (or passed in value on error)
+ */
+export function decodeURIIfValid( encodedURI: string | Stringable ): TypedURL {
+	let encodedURIString: string;
```

Comment:
If I'm not mistaken, all the conditional blocks preceeding the try/catch can be handled right here without type assertions.

```suggestion
	const encodedURIString: string = encodedURI ? encodedURI.toString() : '';
```